proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -98,6 +99,13 @@ public void sendPacket(DefinedPacket packet)
     private boolean legacy;
     @Getter
     private String extraDataInHandshake = "";
+    private boolean disconnecting = false;
```

Comment:
The problem is that the channel close is delayed 500ms, in this time we would then recieve the users ```LoginRequest``` (as its send together).